### ISSUE TITLE
Problem with GHC version during installation

### ISSUE THREAD

**Issue by mwerbos** [begriffs/haskell-vim-now#140]
When doing a clean installation of Stack and Haskell-Vim-Now on a machine with no Haskell-related programs yet installed, I see the following installer output:

~~~~~~~~~~~~~~~~~~~~~~

--> Existing Haskell-Vim-Now installation detected at /home/maia/.config/haskell-vim-now.
--- Syncing Haskell-Vim-Now with upstream...
Current branch master is up to date.
--- Installing system package dependencies...
--- Installing with apt-get...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package fonts-powerline
--- Checking ctags' exuberance...
--- Setting up GHC if needed...
Downloaded lts-4.0 build plan.    
Fetched package index.                                                                                    
Populated index cache.    
Downloaded ghc-7.10.3.                                     
Installed GHC.                              
	Stack bin path: /home/maia/.local/bin
	Stack global path: /home/maia/.stack
	Stack global config location: /home/maia/.stack/global-project/stack.yaml
--- Adding extra stack deps if needed...
--- Installing helper binaries...
Downloaded nightly-2015-12-08 build plan.    
No compiler found, expected minor version match with ghc-7.10.2 (x86_64) (based on resolver setting in /home/maia/.stack/global-project/stack.yaml).
Try running "stack setup" to install the correct GHC into /home/maia/.stack/programs/x86_64-linux/
*** Binary installation failed with error 1. Aborting...
--- Installing git-hscope...
--- Building Hoogle database...
/home/maia/.config/haskell-vim-now/scripts/setup.sh: line 124: /home/maia/.local/bin/hoogle: Not a directory
--- Configuring codex to search in stack...
--- Backing up current vim config using timestamp 20160107_094300...
	/home/maia/.config/haskell-vim-now/backup/.vimrc.20160107_094300
--- Creating symlinks
	~/.vimrc -> /home/maia/.config/haskell-vim-now/.vimrc
	~/.vim   -> /home/maia/.config/haskell-vim-now/.vim
	/home/maia/.config/haskell-vim-now/.stack-bin -> /home/maia/.local/bin
--- Installing vim-plug
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 61054  100 61054    0     0   257k      0 --:--:-- --:--:-- --:--:--  256k
--- Installing plugins using vim-plug...

make -f make_unix.mak
make[1]: Entering directory `/home/maia/.config/haskell-vim-now/.vim/bundle/vimproc.vim'
cc -W -O2 -Wall -Wno-unused -Wno-unused-parameter -std=gnu99 -pedantic -shared -fPIC -o lib/vimproc_linux64.so src/proc.c -lutil
make[1]: Leaving directory `/home/maia/.config/haskell-vim-now/.vim/bundle/vimproc.vim'
 
--- Setting git to use fully-pathed vim for messages...


--- <---- HASKELL VIM NOW installation successfully finished ---->


--> If you are using NeoVim
	Run /home/maia//home/maia/.config/haskell-vim-now/scripts/neovim.sh to backup your existing
	configuration and symlink the new one.
--> Note for a good-looking vim experience:
	Configure your terminal to use a font with Powerline symbols.
	https://powerline.readthedocs.org/en/master/installation.html#fonts-installation

~~~~~~~~~~~~~~~~~~~~~~

Note here that it claims to be expecting ghc-7.10.2 "based on resolver setting in /home/maia/.stack/global-project/stack.yaml", but this is not true. That resolver setting is set to "lts-4.0" by default, which seems to correspond to ghc-7.10.3. It's expecting ghc-7.10.2 because of the `--resolver nightly-2015-12-08` argument in the `stack install` command in the install script.

I was able to fix the problem by modifying ~/.stack/global-project/stack.yaml to be on `resolver: nightly-2015-12-08` (consistent with the stack installation in the install script) rather than `resolver: lts-4.0`, which was the default. Then when I reran the installer, it was forced to install ghc-7.10.2. (I also tried modifying the scripts to use a resolver that had ghc-7.10.3, but this seemed to break some packages.)

However, this seems pretty far from optimal, since I probably don't want to be using the nightly December 8th 2015 build as my global stack resolver.

(Presumably the reason this issue is happening now and didn't before is because the stack default installation has changed recently.)

I *think* the good solution here is to add `--resolver nightly-2015-12-08` to the `stack setup` command in the install script. Will try nuking my installation and doing that, and then send a PR if it solves the issue.

**Comment by begriffs** [comment]
Thanks for reporting the problem. Let me know how it works using that nightly in the setup.

In case you're wondering, the reason I use that nightly snapshot for installation is because of issue #121 (and extra version pinning was required due to #132)

**Comment by xenog** [comment]
Running this command before starting installation of `haskell-vim-now` solves this problem:

```sh
stack setup --resolver nightly-2015-12-08
```

**Comment by jship** [comment]
On OS X 10.11.2, I am encountering this issue.  ```stack setup --resolver nightly-2015-12-08``` does not update ```~/.stack/global-project/stack.yaml```'s resolver to ```nightly-2015-12-08```, so the stack commands run after ```setup``` get confused on GHC versions (even though GHC 7.10.2 was successfully downloaded).

Like @mwerbos mentioned, I can edit the global-project stack.yaml to get it to work.  I have done this via ```sed```.  Will open up a new issue.
